App/Containers/Comments.tsx

```diff
@@ -57,6 +67,24 @@ class Comments extends Component<Props> {
       // New comment added, scroll down, need timeout to allow rendering
       setTimeout(this.scrollToEnd, 100)
     }
+    if (this.props.commentValue === undefined && this.state.submitting === true) {
+      // the comment was flushed so we can type a new one and submit again
+      this.setState({submitting: false})
+    } else if (this.props.commentError === true && this.state.submitting === true) {
+      // there was an error, allow a retry
+      this.setState({submitting: false})
+    }
+  }
+
+  onSubmit = () => {
+    return () => {
```

Comment:
You don't have to do these nested functions. Just eliminate the nested one, and in the `onSubmit` binding below, make it like: `onSubmit={this.onSubmit}`

Reply:
fixed